soc/silabs/silabs_s2/config/sl_device_init_dcdc_config.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SL_DEVICE_INIT_DCDC_CONFIG_H
+#define SL_DEVICE_INIT_DCDC_CONFIG_H
+
+#include <zephyr/devicetree.h>
+#include <soc.h>
+
+#if DT_HAS_COMPAT_STATUS_OKAY(silabs_series2_dcdc)
+
+#define DCDC_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(silabs_series2_dcdc)
+
+#define SL_DEVICE_INIT_DCDC_ENABLE DT_PROP(DCDC_NODE, regulator_boot_on)
+#define SL_DEVICE_INIT_DCDC_BYPASS DT_PROP(DCDC_NODE, regulator_allow_bypass)
+
+#define SL_DEVICE_INIT_DCDC_TYPE DT_PROP_OR(DCDC_NODE, regulator_initial_mode, 0)
+
+#define SL_DEVICE_INIT_DCDC_BOOST_OUTPUT                                                           \
+	(DT_ENUM_HAS_VALUE(DCDC_NODE, regulator_init_microvolt, 1800000)                           \
+		 ? emuDcdcBoostOutputVoltage_1v8                                                   \
+	 : DT_ENUM_HAS_VALUE(DCDC_NODE, regulator_init_microvolt, 1900000)                         \
+		 ? emuDcdcBoostOutputVoltage_1v9                                                   \
+	 : DT_ENUM_HAS_VALUE(DCDC_NODE, regulator_init_microvolt, 2000000)                         \
+		 ? emuDcdcBoostOutputVoltage_2v0                                                   \
+	 : DT_ENUM_HAS_VALUE(DCDC_NODE, regulator_init_microvolt, 2100000)                         \
+		 ? emuDcdcBoostOutputVoltage_2v1                                                   \
+	 : DT_ENUM_HAS_VALUE(DCDC_NODE, regulator_init_microvolt, 2200000)                         \
+		 ? emuDcdcBoostOutputVoltage_2v2                                                   \
+	 : DT_ENUM_HAS_VALUE(DCDC_NODE, regulator_init_microvolt, 2300000)                         \
+		 ? emuDcdcBoostOutputVoltage_2v3                                                   \
+	 : DT_ENUM_HAS_VALUE(DCDC_NODE, regulator_init_microvolt, 2400000)                         \
+		 ? emuDcdcBoostOutputVoltage_2v4                                                   \
+		 : emuDcdcBoostOutputVoltage_1v8)
```

Comment:
aren't enums correlated to voltages so that something like linear range can be used? This looks seriously ugly.